pkg/common/cluster_state.go

```diff
@@ -24,19 +24,19 @@ func NewClusterState() *ClusterState {
 	}
 }
 
-func (i *ClusterState) Read(cr *kc.Keycloak, controllerClient client.Client) error {
-	return i.readKeycloakServiceCurrentState(cr, controllerClient)
+func (i *ClusterState) Read(context context.Context, cr *kc.Keycloak, controllerClient client.Client) error {
```

Comment:
3 arguments is my personal max. I believe we'll need to refactor this in the direction @pb82 was suggesting (inject those objects in a constructor and then call unparametrized `Read`. But let's leave it for now, we can easily refactor this later.